### DISCUSSION TITLE
Trying to port the zynq_timestamping solution to work with zynq ADRV9361-Z7035 picozed sdr bob board.

### DISCUSSION THREAD

**Discussion by xilinx-seeker** [srsran/zynq_timestamping#15] [General] [answered]
I have integrated the zynq_timestamping IPs with the ADI firmware solution for the ADI adrv bob board. I am getting the following error during the place & route stage of bitstream generation.

**_**__**_**[DRC INBB-3] Black Box Instances: Cell 'i_system_wrapper/system_i/dac_fifo_timestamp_e_0/U0/GEN_MEMORY_CH01[0].PS_DAC_data_RAMB_ch01_0_inst/ramb36e2_gen.inBuffer_mem_13_inst' of type 'system_dac_fifo_timestamp_e_0_0_RAMB36E2' has undefined contents and is considered a black box.  The contents of this cell must be defined for opt_design to complete successfully. (99 more like this).**

Please suggest a possible solution for the above problem.

Thanks in advance.

**Comment by xilinx-seeker** [comment, marked as answer]
The problem is solved by initializing the ramb36e1 in the dac_fifo_timestamp_enabler RTL code for zynq-based devices.

**Reply by ofontbach** [reply]
@xilinx-seeker, please be aware that the wrong ramb primitive wrapping file was initially pushed, which effectively means that the ramb36e1 primitive was indeed never used. This has been corrected and we encourage you to update your local repository and try again. Thanks!
